core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -905,19 +908,35 @@ class GroupCoordinator(val brokerId: Int,
    *
    * @param offsetTopicPartitionId The partition we are now leading
    */
-  def onElection(offsetTopicPartitionId: Int): Unit = {
-    info(s"Elected as the group coordinator for partition $offsetTopicPartitionId")
-    groupManager.scheduleLoadGroupAndOffsets(offsetTopicPartitionId, onGroupLoaded)
+  def onElection(offsetTopicPartitionId: Int, coordinatorEpoch: Int): Unit = {
+    epochForPartitionId.compute(offsetTopicPartitionId, (_, epoch) => {
+      val currentEpoch = Option(epoch)
+      if (currentEpoch.forall(currentEpoch => coordinatorEpoch > currentEpoch)) {
```

Comment:
One final thing I was considering is whether we should push this check into `GroupMetadataManager.loadGroupsAndOffsets`. That would give us some protection against any assumptions about ordering in `KafkaScheduler`.